### PR TITLE
Cobbduceus Part 2/X: Medibot-chan Tends Your Wounds

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
### Medibot Tend Wounds
Medibot has been reworked to perform a baby version of tend wounds. It still heals all base damage types.

Medibots now scale with tend wound surgeries researched. Each one increases HPS by 50%. Requires a "Refresh" similar to Operational tables so don't ignore them!

### Medibot Beaker/Virus Removal
Medibots no longer hold beakers (because they do a pretend surgery) nor heal viruses (because they do a pretend surgery)

### Misc
Using a niche medkit will allow the bot to be more efficient in healing that damage, increasing their HPS post techweb-scaling by 50%

Emagged borgs now give you 5u chloral and damage you based on what they would have healed (brute if you don't need treatment) per step.

### Medibot Tend Wounds
because passive healing is lame. Post C2 additions, medibots are really not safe to have around. These revert them back to the little helpers they used to be without making them a (good) substitute for doctors.

Also means we can balance these things separately from chems. No more "THINK OF THE SILICONS!"... well at least for this guy.

Requires a refresh since I don't want it constantly checking for tech nor do I want people to just fire/forget the things.

### Medibot Beaker/Virus Removal
Side effect of Tend Wounds since the bot no longer is reagent-oriented. Doctors should be handling virus outbreaks anyways, not bots.

### Misc
neat feature a la bee suit protecting from bees.

I needed to give the ughmag something that made sense with the change, and I think this buffs it in stealth and power if you aren't quick to react since it'll put you to sleep. Fear the emagged medibot!

## Changelog
:cl: Medibotby
balance: Medibot no longer works off chems and instead performs a lesser equivalent to Tend wounds
balance: They still heal all 4 base damages and use same clothing restrictions as previously
del: Medibot Beaker Holding / Virus Treating no longer exists
add: Medibots scale with Tend Wound research (1 tend wound surgery researched = 50% more healing)
add: Medibots heal more of a specific damage if the medkit used to make them was oriented around that damage (a tox medkit medibot will heal toxin better)
tweak: Emagged Medibots now pretend to tend your wounds, actually damaging you and applying 5u chloral per step.
/:cl:

### CODE REVIEW

**Comment by Dawson1917** [comment]
image
image

**Comment by PKPenguin321** [comment]
Give him a spray bottle of that chem that makes surgery safer and keep the beaker functionality so you can still subvert him by filling it with acid

**Comment by carshalash** [comment]
>No more "THINK OF THE SILICONS!"... well at least for this guy.

Fucking personal attack, smh.

It is nice that these things are no longer break on sight though.

**Comment by CRITAWAKETS** [comment]
Why do you name your PRs like it was a dyslexic trying to create new words while having an epileptic attack.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Give him a spray bottle of that chem that makes surgery safer and keep the beaker functionality so you can still subvert him by filling it with acid

I have never seen someone use this feature which is why I didn't shed tears dropping it in the first place. I'll try to think of a subversion mechanic tho.

Also don't tell anyone but she doesn't actually do surgery.

**Comment by Shapsy** [comment]
You should make it so they put a little heart bandaid on u when they're done :)

**Comment by PKPenguin321** [comment]
>I'll try to think of a subversion mechanic tho that isn't bound to the cringemag.

I just told u one >:(

**Comment by Tlaltecuhtli** [comment]
how do multiple medibots work on the same person?
can they perform only the head is exposed?

**Comment by ATH1909** [comment]
Can they heal unexposed parts of your body? What if another part of your body is exposed?

Also, why do medibots get to do tox/oxy-healing surgeries (y'know, ones that would actually be USEFUL/USED (well, mostly just the tox one)) that us humanoids still don't have access to? I DEMAND a tox damage tend wounds surgery!

Another potential concern is that no one is gonna do the tend wounds surgeries themselves when they can just strap a patient to a stasis bed next to a medibot and have the medibot do it for them, especially if the medibot bypasses most forms of clothing and such (unlike "real" surgery).

**Comment by ExcessiveUseOfCobblestone** [comment]
> how do multiple medibots work on the same person?
> can they perform only the head is exposed?

they mega surgery

**Comment by ExcessiveUseOfCobblestone** [comment]
> Can they heal unexposed parts of your body? What if another part of your body is exposed?

Yes, if they could inject you previously they can tendwounds. It is not bodypart specific.

> Also, why do medibots get to do tox/oxy-healing surgeries (y'know, ones that would actually be USEFUL/USED (well, mostly just the tox one)) that us humanoids still don't have access to? I DEMAND a tox damage tend wounds surgery!
> 
> Another potential concern is that no one is gonna do the tend wounds surgeries themselves when they can just strap a patient to a stasis bed next to a medibot and have the medibot do it for them, especially if the medibot bypasses most forms of clothing and such (unlike "real" surgery).

the surgery is considerably lower so I mean yeah you CAN do it but yikes just use cryo if you're gonna do that.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'll prob need to lower healing output and add something that dissuades just spamming the buggers.

**Comment by shellspeed1** [comment]
> 
> 
> I'll prob need to lower healing output and add something that dissuades just spamming the buggers.

Just make it apply slowdown after treatment, should work well enough.

**Comment by actioninja** [comment]
cobby I am going to throw you out a window if you keep these names up

**Comment by Akrilla** [comment]
`tweak: Emagged Medibots now pretend to tend your wounds, actually damaging you and applying 5u chloral per step.` wouldn't this be really obvious? Maybe they should just do nothing but apply the chemical?

**Comment by Akrilla** [comment]
also ExcessiveUseOfCobblestone:hewwo-its-me-mediwoni 👀

**Comment by dr-bee** [comment]
Mood penalty would be a good downside in general. 

Also dont let them stack, only allow one medibot to work on one target at a time. Another method of balance would be to only able to heal people who are outside of crit.

Below crit levels they apply epinephrine and scream for help. This makes them handy for boo-boos but you still need doctors to get people out of crit.

**Comment by Striders13** [comment]
Will pAI controlled medibots work with this?

**Comment by Tlaltecuhtli** [comment]
does a medibot made with advanced medkit give you anti stun surgery while healing you?

**Comment by ExcessiveUseOfCobblestone** [comment]
> `tweak: Emagged Medibots now pretend to tend your wounds, actually damaging you and applying 5u chloral per step.` wouldn't this be really obvious? Maybe they should just do nothing but apply the chemical?

I consider it the same level of stealth (or lack their of) associated pre-pr which was "OH I WANT TO INJECT EVERYONE". The difference here is that you need to kill it or gtfo before it KOs you then slowly hecks you up. It will also sometimes do different damagetypes depending on third party stuff (IE it first treats brute but eventually your tox outruns your brute because of some other poison, it will switch to more tox).

**Comment by ExcessiveUseOfCobblestone** [comment]
> cobby I am going to throw you out a window if you keep these names up

I was going to name it "oniichan what are you going to do with that robotic arm, b-b-b-baka!"

**Comment by Fikou** [comment]
do bezerk medbots have any buffs

**Comment by ExcessiveUseOfCobblestone** [comment]
they heal more @ base meaning they'll also have more HPS/tech.

**Comment by optimumtact** [comment]
is this ready?

**Comment by ExcessiveUseOfCobblestone** [comment]
I want to inhibit multi-botting and maybe look at the numbers a bit more, but otherwise yes.

I will finish this evening EST

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -146,6 +146,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_OIL_FRIED			"oil_fried"
 #define TRAIT_MEDICAL_HUD		"med_hud"
 #define TRAIT_SECURITY_HUD		"sec_hud"
+#define TRAIT_MEDIBOTCOMINGTHROUGH "medbot" //Is a medbot healing you
```
Comment (Akrilla) [inline code/__DEFINES/traits.dm]: we like weirdly named defines

Comment (ExcessiveUseOfCobblestone) [reply code/__DEFINES/traits.dm]: you'll never forget what it does

**Pull request merged**